lib/optimizely.rb

```diff
@@ -109,8 +97,8 @@ def activate(experiment_key, user_id, attributes = nil)
       end
 
       # Create and dispatch impression event
-      experiment = @config.get_experiment_from_key(experiment_key)
-      send_impression(experiment, variation_key, user_id, attributes)
+      experiment = config.get_experiment_from_key(experiment_key)
+      send_impression(config, experiment, variation_key, user_id, attributes)
```

Comment:
Does Ruby offer any mutex / locking mechanisms so we can avoid passing this around?